src/msw/listctrl.cpp

```diff
@@ -3275,20 +3280,64 @@ void wxListCtrl::OnPaint(wxPaintEvent& event)
         return;
     }
 
+    int visibleEnd = countPerPage;
+    visibleEnd -= 1;
+
     const long top = GetTopItem();
     const long bottom = wxMin(top + countPerPage, itemCount - 1);
 
+    wxRect clipRect;
+    dc.GetClippingBox(clipRect);
+
+    if (m_listRulesAlternateColourOnBlank && visibleEnd > bottom ) {
+        wxPaintDC dc2(this);
+        wxRect itemRect;
+        if (GetItemRect(0, itemRect)) {
+            dc2.SetBrush(m_alternateColourOnBlank);
+            dc2.SetPen(*wxTRANSPARENT_PEN);
+            wxRect rect;
+            int iniy = clipRect.y;
+            rect.x = 0;
+            rect.y = clipRect.y;
+            rect.width = clipRect.width;
+            rect.height = itemRect.height;
+            for ( long i = bottom+1; i <= visibleEnd + 1 ; i++ )
+            {
+                if ( i % 2 ) {
+                    rect.y = iniy + (i * itemRect.height);
+                    dc2.DrawRectangle(rect);
+                }
+            }
+
+            // Need to include 4 lines below to update painting of
+            // Blank Lines (Unfortunately theres is some flickering
+            if ( iRefresh == 0 ) {
+                Refresh();
```

Comment:
Sorry for the stupid question but how is Refresh() supposed to work when called inside the drawing function, won't this just trigger another paint event?